### PR TITLE
add feature upload xlsx in existing table

### PR DESCRIPTION
**Pull request by semmjon**
## Change Summary

PR for:
- closes https://github.com/nocodb/nocodb/issues/5021
- closes https://github.com/nocodb/nocodb/issues/4271

 I added the feature that enables users to upload Excel file templates and import data into an existing table. I accomplished this by modifying the QuickImport.vue file and adding some code to handle the uploading of the new file types. I also updated the ViewActions.vue file to display the new upload options in the project menu, and created a new QuickImportDialogType to differentiate between CSV, Excel, and JSON import options. Users will now be able to easily import data from Excel and JSON files into an existing table.

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

The first file, packages/nc-gui/components/dlg/QuickImport.vue, was changed to add support for uploading XLSX and JSON files into an existing table. Specifically, the following changes were made:

importState was changed from being an object created using reactive to a defaultImportState object. The importState variable is now created by making a reactive copy of defaultImportState. This is a best practice when using the reactive API.
A new defaultImportState object was created that defines default values for the fileList, url, jsonEditor, jsonSchema, delimiter, firstRowAsHeaders, and shouldImportData properties.
The handleImport function was modified to set templateEditorModal to false and to reset the importState to its default values after the import is complete.
The second file, packages/nc-gui/components/smartsheet/toolbar/ViewActions.vue, was modified to add support for uploading XLSX and JSON files. Specifically, the following changes were made:

A new QuickImportDialogType type was defined that specifies the different types of import dialogs that can be displayed.
A new quickImportDialogTypes array was defined that contains the different import dialog types.
A new quickImportDialogs object was defined that maps import dialog types to ref variables that control whether the dialog is displayed or not.
The a-menu-item template was modified to loop over each type of import dialog and display a menu item for each.
A new LazyDlgQuickImport component was added to the file. This component takes a v-model prop that controls whether the component is displayed or not. The component is used to display the import dialog when the corresponding menu item is clicked.

Tested locally in a dev-scenario (npm run dev / run:watch) and in a deployed scenario with a dockerfile / helm-chart.

### CODE REVIEW

**Review by wingkwong — CHANGES_REQUESTED** [review]
Uploading a JSON file (either with correct mapping and incorrect one) would cause forever loading with a successful message while it is failing behind the scene. 

image

You may try csv or excel to see the expected behaviours if there is no correct mapping, namely

- the destination column is not mapped 
- the source column is not checked

image

Otherwise, it should show the correct mapping. Here's the sample json. 

```json
[
    {
        "country": "Country 1",
        "lang": "Language 1",
        "foo": "Bar"
    },
    {
        "country": "Country 2",
        "lang": "Language 2",
        "foo": "Bar"
    }
]
```

**Comment by semmjon** [comment]
Sry, i dont testet json before. Maybe we'll remove json from the list for now until I have time to test it?

**Comment by wingkwong** [comment]
You may remove Json one first as I think it requires some extra implementation.

**Comment by wingkwong** [comment]
whenever it is ready, please click re-request review button to notify me to have a review again.

File: packages/nc-gui/components/smartsheet/toolbar/ViewActions.vue [hunk 1]
```diff
@@ -230,7 +241,14 @@ useMenuCloseOnEsc(open)
       </template>
     </a-dropdown>
 
-    <LazyDlgQuickImport v-if="quickImportDialog" v-model="quickImportDialog" import-type="csv" :import-data-only="true" />
+    <LazyDlgQuickImport
+      v-for="type in quickImportDialogTypes"
+      :key="type"
+      v-model="quickImportDialogs[type].value"
+      :import-type="type"
+      :base-id="bases[0].id"
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/toolbar/ViewActions.vue:249]: A project can include different bases. You should pass the one that the opened table belongs to.

File: packages/nc-gui/components/smartsheet/toolbar/ViewActions.vue [hunk 2]
```diff
@@ -41,11 +42,20 @@ const showApiSnippetDrawer = ref(false)
 
 const showErd = ref(false)
 
-const quickImportDialog = ref(false)
+type QuickImportDialogType = 'csv' | 'excel' | 'json'
+// TODO: add 'json' when it's ready
+const quickImportDialogTypes: QuickImportDialogType[] = ['csv', 'excel']
+const quickImportDialogs: Record<typeof quickImportDialogTypes[number], Ref<boolean>> = quickImportDialogTypes.reduce(
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/toolbar/ViewActions.vue:48]: add line breaks between them.

**Review by wingkwong — CHANGES_REQUESTED** [review]

**Review by wingkwong — CHANGES_REQUESTED** [review]
Still not catering https://github.com/nocodb/nocodb/pull/5164#discussion_r1113886062

**Review by wingkwong — APPROVED** [review]

**Pull request merged**